index.js

```diff
@@ -1 +1,96 @@
 // implement your API here
+const express = require('express');
+const db = require('./data/db.js');
+
+const server = express();
+server.use(express.json());
+
+//adds a user to the database
+//user must include the fields name and bio
+server.post('/api/users', (req, res) => {
+    const { name, bio } = req.body;
+    const user = req.body;
+
+    if(!name || !bio) {
+        return res.status(400).json({error: "Please provide name and bio for the user." });
+    }
+    
+    db.insert(user)
+        .then(user => {
+            res.status(201).json(user);
+        }).catch(error => {
+            res.status(500).json({ error: "There was an error while saving the user to the database" });
+        });
+
+});
+
+//retrieves all users in the database
+server.get('/api/users', (req, res) => {
+    db.find()
+        .then(users => {
+            res.status(200).json(users);
+        }).catch(error => {
+            res.status(500).json({ error: "The users information could not be retrieved." });
+        });
+});
+
+//finds a specific user by id within the database
+//returns undefined if the user does not exist
+server.get('/api/users/:id', (req, res) => {
+    const id = req.params.id;
+
+    db.findById(id)
+        .then(user => {
+            console.log(user);
+            if(user){
+                return res.status(200).json(user);
+            } else {
```

Comment:
You don't actually need the else, you just need the if to check and if it doesn't pass then return you can send what you expect